oeps/oep-0028-content-theming-in-react.rst

```diff
@@ -0,0 +1,246 @@
+|                 |                                                                |
+|                 |                                                                |
+|                 |                                                                |
++-----------------+----------------------------------------------------------------+
+| Title           | Content Theming in React                                       |
++-----------------+----------------------------------------------------------------+
+| Last Modified   | 2018-08-24                                                     |
++-----------------+----------------------------------------------------------------+
+| Authors         | Braden MacDonald <braden@opencraft.com>,                       |
+|                 | Taranjeet Singh <taranjeet@opencraft.com>                      |
++-----------------+----------------------------------------------------------------+
+| Arbiter         | Ari Rizzitano <arizzitano@edx.org>                             |
++-----------------+----------------------------------------------------------------+
+| Status          | Draft                                                          |
++-----------------+----------------------------------------------------------------+
+| Type            | Architecture                                                   |
++-----------------+----------------------------------------------------------------+
+| Created         | 2018-08-17                                                     |
++-----------------+----------------------------------------------------------------+
+| `Review Period` | 2018-08-17 - 2018-09-02                                        |
++-----------------+----------------------------------------------------------------+
+
+Abstract
+-------
+
+OEP-0023 [1] addresses the styling part of a theming system based on React. This proposal adds guidelines for content modification aspect to the React based theming system
+
+Context
+-------
+
+Open edX instances frequently wish to customize not only the colors and styles of the Open edX UI, but also to add, remove, and modify significant parts of the UI to suit their use cases and branding. Doing this in a way that is understandable, clean, and easy to maintain can be a challenge unless the original UI is designed with this in mind.
+
+Another pain of theming system is providing access to unrelated data to any component. This data may not be used by earlier components but is required in a particular theme. It becomes an important concern for writing a theming system where a global access to data to all the components in the UI system should be provided.
+
+Decision
+--------
+
+* We will build small modular components which will be non-customizable, stateless and contain HTML, and certain logic. We can see an example of a smaller component SiteLogo, which will later be used in a customizable component _Header.
```

Comment:
It's not really explained here that there are meant to be two types of components: internal/non-customizable and customizable ones.

At first read, this sentence just sounds like "we will build everything out of small components that cannot be customized."